explainers/MultiViewport/explainer.md

```diff
@@ -0,0 +1,112 @@
+
+## Goals
+
+1. Primary goals include increasing the number of users who use enable DeviceMode and increase number of users who test more than 1 device.
+1. Improve satisfaction and efficiency in designing a website through easy access to and visibility of multiple viewports.
+1. Metrics to measure success:
+    1. Session length with DeviceMode enabled (1D vs 2D v 10D user).
+    1. Number of users who enable DeviceMode.
+
+## Non-goals
+
+- This feature will not change the default experience of the current Device Mode in DevTools. This feature will be additive/optional.
+- For the initial implementation of this feature, we will only support 2 viewports, but we are designing the feature to scale up to N viewports in the future.
+
+## Hypotheses
+
+- We believe that using DeviceMode with multiple viewports will increase efficiency in DevTools customers. Additionally, it will increase user engagement either in session length or number of sessions. Web Developers can be more efficient using DeviceMode if they can view their web pages in multiple screen sizes, dimensions, and device forms side-by-side in DevTools. 
+- The current single viewport device mode has valuable screen real estate that is being underutilized and we think that we can bring in new functionality by adding multiple viewports without negatively changing the experience and functionality today. As of right now, multiple viewports are available but only when using extensions or multiple tools. Adding multiple viewports directly into DevTools will streamline the user experience.
+
+## Experimentation
+
+We know this feature will be shipped as an experiment in the Microsoft Edge DevTools.
+
+### Trigger Conditions and Treatment
+
+- Opening Device Mode.
+- Showing the option within the dropdown menu to add a new viewport.
+
+### Metrics
+
+- Through this experiment, session times will be analyzed.
+- Additionally, (for the treatment group) this experiment will provide further insight into how many screens are needed by users to be productive. This can be tracked by checking how many screens are activated in each working session.
+
+## Usage
+
+### Add a new viewport
+
+Users will be able to add a new device to the viewport by simply checking a box from the dropdown list of devices in the viewport toolbar. Once a device is added, it will appear either next to or underneath the existing viewport(s) with a toolbar on top of it. The devices will be re-centered to accommodate the new viewport. As more devices are added and the viewport fills up, the user will be able to vertically scroll down the viewport to look at all the devices.
+
+### Open DevTools against a new viewport
+
+DevTools will remain the same, regardless of the number of devices in the viewport. As in the current version of DevTools the user will be able to choose the placement of DevTools on their screen and expand/shrink the DevTools window.
```

Comment:
I think there needs to be a note about which viewport is DevTools targeting.

Indeed, depending on the screen size, a webpage may behave differently, have different JS code running, or even have different DOM elements. 
Here are 2 examples:

1. If I add some JS code that sets a variable based on the window size like: `window.a = window.innerWidth > 1000 ? true : false;`. In this case, if I have 2 viewports, one that's narrower than 1000px and the other one wider, then the value of `a` will be `false` in the first case and `true` in the second. So if I use the Console panel in DevTools to log the value of `a`, what should happen? Should the expression be evaluated in all viewports, and all results shown? Or should it be evaluated only in 1 viewport, and in which case, which one?

2. If I have some CSS code that displays a pseudo-element depending on the window size, like `@media (min-width: 1000px) { body::before { content: "Large"; } }`. In this case, if I have 2 viewports (one narrow, one wide), a pseudo-element will exist in one, and not in the other one. So what should be displayed in the Elements panel when I expand the `<body>` node? Should there be a way to see both states at the same time? Or just one state?

The above examples are arbitrary and simple, just to make the point, but these things do happen often in real websites and can be way more complicated.

I think DevTools should target just one viewport, to make things simpler. But then that also raises the question of how changes made in DevTools are replicated. If you start changing CSS in the Styles panel, should the change be mirrored in all other viewports? What about if you change the DOM (modifying an attribute, or adding a node), etc.

There are probably multiple steps that can be taken in achieving this, but the explainer should mention it.

Reply:
noted, thank you for the scenarios -- we will definitely be thinking about them as a team throughout this project!

Reply:
Sounds good. Because this can be potentially very complex, I would advise starting with DevTools being attached to only one viewport, i.e. the first one. Secondary viewports that get added after this would be synchronized as you explained, but would not be reachable with DevTools.
I think it is important to add a note about this to the explainer @hanamemon.

Reply:
I agree -- updating the explainer now to reflect this perspective.